packages/runtime/src/runtime.ts

```diff
@@ -414,8 +441,21 @@ export class Runtime {
       args: args
     };
 
-    await this.run(program);
-    this.updateFinalState(txnParams); // update account balances
+    const txnParameters = Array.isArray(txnParams) ? txnParams : [txnParams];
+
+    try {
+      await this.run(program);
+    } catch (error) {
+      // if transaction type is Clear Call, remove the app first before throwing error (rejecting tx)
```

Comment:
in `clear` does app's local state is deleted irrespective to rejection or acception ? if so please add a link to algorand reference

Reply:
Yes. Ok will add a link